### DISCUSSION TITLE
How to define interval that snmp_exporter queries targets?

### DISCUSSION THREAD

**Discussion by lonelyadmin** [prometheus/snmp_exporter#839] [Q&A] [answered]
Some of my older snmp targets can't handle being queried every 15s. I verified they're getting queried every 15s by looking at tcpdump and log files.

I still consider myself new to prometheus, but my understanding is that the actual querying of the snmp device is defined at the exporter level and not at the prometheus level. the global part of the prometheus config does have scrape_interval and evaluation_interval both set to 15s...but that's just how often it checks the metrics provided by snmp_exporter...right?

How can I change the querying interval from snmp_exporter to devices both globally and per device? And if I change a device to something longer, say 5m, do I need to change prometheus to not scrape more often than 5m for that one device? Or is it's scrape_interval only configurable for the entire exporter/job?

Or am I entirely off in the woods....? 

Thanks!

**Comment by SuperQ** [comment, marked as answer]
Scrape intervals are defined in Prometheus, not the exporter. The exporter collects when pulled.

**Reply by lonelyadmin** [reply]
Is the pulling/querying interval by the exporter configurable? Or is it acting as a sort of proxy? It only queries the device when prometheus scrapes it? If that's the case, is the amount of bulk queries sent configurable per device? I think I found this referenced as config.WalkParams.MaxRepetitions in the source, which I think can be set as max_repetitions: int in the generator config, but can this only be set for a given target? I don't want all of the targets to share the same settings.

I have these old firewalls that can only be queried slowly. Knowing how to lower the interval and how much is queried in the bulk request is the only way I can get reliable data from them.

Thanks for your time, and patience.

**Reply by nemobis** [reply]
@lonelyadmin You can increase the `scrape_interval` in your [scrape_configs](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#scrape_config). If Prometheus scrapes snmp_exporter less often, less requests will be made.
